extensions/b2g/viewer.js

```diff
@@ -0,0 +1,214 @@
+/* Copyright 2014 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* globals PDFJS, Promise */
+
+'use strict';
+
+PDFJS.useOnlyCssZoom = true;
+PDFJS.disableTextLayer = true;
+PDFJS.workerSrc = '../pdfjs-components/build/pdf.worker.js';
+PDFJS.cMapUrl = '../pdfjs-components/cmaps/';
+PDFJS.cMapPacked = true;
+
+var DEFAULT_SCALE_DELTA = 1.1;
+var MIN_SCALE = 0.25;
+var MAX_SCALE = 10.0;
+
+var PDFViewerApplication = {
+  pdfDocument: null,
+  pdfViewer: null,
+  pdfHistory: null,
+  pdfLinkService: null,
+  loading: true,
+  updateScaleControls: false,
+
+  open: function (params) {
+    var url = params.url, originalUrl = params.originalUrl;
+    this.setTitleUsingUrl(originalUrl);
+
+    // Loading document.
+    PDFJS.getDocument(url).then(function (pdfDocument) {
+      // Document loaded, specifying document for the viewer.
+      this.pdfDocument = pdfDocument;
+      this.pdfViewer.setDocument(pdfDocument);
+      this.pdfLinkService.setDocument(pdfDocument);
+      this.pdfHistory.initialize(pdfDocument.fingerprint);
+    }.bind(this), function (error) {
+      // TODO handle and localize loading error?
+      console.error('PDF loading error: ' + error)
```

Comment:
We're missing a semicolon here at the end (Travis also appears to be complaining about that).